modules/payment/src/Plugin/Commerce/CheckoutPane/PaymentInformation.php

```diff
@@ -122,8 +126,57 @@ public function buildPaneForm(array $pane_form, FormStateInterface $form_state,
     // @todo Support multiple gateways.
     /** @var \Drupal\commerce_payment\Entity\PaymentGatewayInterface $payment_gateway */
     $payment_gateway = reset($payment_gateways);
+
+    $pane_form['payment_gateway'] = [
+      '#type' => 'value',
+      '#value' => $payment_gateway->id(),
+    ];
+
     $payment_gateway_plugin = $payment_gateway->getPlugin();
 
+    if ($payment_gateway_plugin instanceof SupportsStoredPaymentMethodsInterface) {
+      $this->attachPaymentMethodForm($payment_gateway, $pane_form, $form_state);
+    }
+    else {
+      /** @var \Drupal\profile\Entity\ProfileInterface $billing_profile */
+      $billing_profile = $this->order->getBillingProfile();
+      if (!$billing_profile) {
+        $billing_profile = Profile::create([
+          'uid' => $this->order->getCustomerId(),
+          'type' => 'customer',
+        ]);
+      }
+
+      $form_display = EntityFormDisplay::collectRenderDisplay($billing_profile, 'default');
```

Comment:
We keep repeating this bit, here, on the billing information pane, soon for shipping. We need to abstract it out somehow/somewhere.